streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalProcessorContextImpl.java

```diff
@@ -47,23 +50,48 @@ public StateStore getStateStore(final String name) {
     @SuppressWarnings("unchecked")
     @Override
     public <K, V> void forward(final K key, final V value) {
+        forward(key, value, SEND_TO_ALL);
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public <K, V> void forward(final K key, final V value, final To to) {
+        toInternal.update(to);
+        if (toInternal.hasTimestamp()) {
+            recordContext.setTimestamp(toInternal.timestamp());
+        }
         final ProcessorNode previousNode = currentNode();
         try {
-            for (final ProcessorNode child : (List<ProcessorNode<K, V>>) currentNode().children()) {
-                setCurrentNode(child);
-                child.process(key, value);
+            final List<ProcessorNode<K, V>> children = (List<ProcessorNode<K, V>>) currentNode().children();
```

Comment:
Thinking about this once more, can a "global processor" have multiple children to begin with? Maybe we can simplify this method significantly? Thoughts?

Will add a test for `GlobalProcessorContextImpl` though -- there is not at all atm.

Reply:
> Thinking about this once more, can a "global processor" have multiple children to begin with? 

Yeah, you could be correct about, I don't think it can.